Stratis.Bitcoin.Features.Miner/Controllers/MinerController.cs

```
@@ -0,0 +1,118 @@
+                if (!this.ModelState.IsValid)
+                {
+                    var errors = this.ModelState.Values.SelectMany(e => e.Errors.Select(m => m.ErrorMessage));
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, "Formatting error", string.Join(Environment.NewLine, errors));
+                }
+
+                GetStakingInfoModel model = this.posMinting != null ? this.posMinting.GetGetStakingInfoModel() : new GetStakingInfoModel();
+
+                return this.Json(model);
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, e.Message, e.ToString());
+            }
+        }
+
+        /// <summary>
+        /// Start staking.
+        /// </summary>
+        /// <param name="request">The name and password of the wallet to stake.</param>
+        /// <returns>An OKResult object that produces a status code 200 HTTP response.</returns>
+        [Route("startstaking")]
+        [HttpPost]
+        public IActionResult StartStaking([FromBody]StartStakingRequest request)
+        {
+            try
+            {
+                if (!this.ModelState.IsValid)
+                {
+                    var errors = this.ModelState.Values.SelectMany(e => e.Errors.Select(m => m.ErrorMessage));
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, "Formatting error", string.Join(Environment.NewLine, errors));
+                }
+
+                WalletManager walletManager = this.fullNode.NodeService<IWalletManager>() as WalletManager;
+
+                Wallet.Wallet wallet = walletManager.Wallets.FirstOrDefault(w => w.Name == request.Name);
+
+                if (wallet == null)
+                {
+                    wallet = walletManager.LoadWallet(request.Password, request.Name);
```

I don't think you need to load the wallet, this should happen automatically on startup.
If the wallet is not found then its a wrong name so we can throw an exception.

---

Ok done.